lib/IDE/Refactoring.cpp

```diff
@@ -7363,23 +7449,71 @@ class AsyncConverter : private SourceEntityWalker {
       OS << tok::kw_try << " ";
     }
     OS << "await ";
-    addRange(CE->getStartLoc(), CE->getFn()->getEndLoc(),
-             /*ToEndOfToken=*/true);
 
-    OS << tok::l_paren;
-    size_t realArgCount = 0;
-    for (size_t I = 0, E = Args.size() - 1; I < E; ++I) {
-      if (isa<DefaultArgumentExpr>(Args[I]))
+    // If there's no alternative then continue, but assume that the alternative
+    // has the same name as the callee
+    if (!HandlerDesc.Alternative) {
+      addRange(CE->getStartLoc(), CE->getFn()->getEndLoc(),
+               /*ToEndOfToken=*/true);
+    } else {
+      // Replace the original call with the name of the alternative
+      Identifier Name;
+      if (auto *Accessor = dyn_cast<AccessorDecl>(HandlerDesc.Alternative))
+        Name = Accessor->getStorage()->getBaseIdentifier();
+      else
+        Name = HandlerDesc.Alternative->getBaseIdentifier();
```

Comment:
Just occurred to me that `getBaseIdentifier` would hit an assertion if this a decl with a special name such as `init`, and I *believe* we support marking those async.

Reply:
`init` can definitely be marked async, but it wouldn't make sense as a completion handler function since it inherently returns a value (and thus should never be an alternative).

Regardless, it's probably better to check and fallback anyway - only adds a line or two.

Reply:
Ah right yeah, sounds good 👍